core/src/main/scala/codesearch/core/index/repository/SourcePackage.scala

```diff
@@ -0,0 +1,63 @@
+package codesearch.core.index.repository
+
+import sys.process._
+import java.io.File
+import java.net.{URI, URLEncoder}
+import java.nio.file.{Path, Paths}
+
+import codesearch.core.util.Helper
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+trait Extractor {
+  def extract(archive: File, directory: File): Future[File] =
+    Future {
+      Seq("tar", "-xvf", archive.getCanonicalPath, "-C", directory.getCanonicalPath) !!;
+      directory
+    }
+}
+
+trait SourcePackage extends Extractor {
+  def fsPath: Path
+  def url: URI
+  def extentions: Set[String] = Set.empty
+}
```

Comment:
Comments please (if you don't know what these fields mean then ask around)

Reply:
It is value by default for trait. Implementations override this field.

Reply:
I meant comments like "`extentions` specifies which files from the package will be retained after the package is downloaded; this is useful to filter out binaries and other artifacts that occasionally occur in package archives"

btw, why is it a part of the trait at all? shouldn't it be per-language, not per-package?